erpnext/manufacturing/doctype/bom_update_log/bom_update_log.py

```diff
@@ -212,7 +212,7 @@ def resume_bom_cost_update_jobs():
 			["name", "boms_updated", "status"],
 		)
 		incomplete_level = any(row.get("status") == "Pending" for row in bom_batches)
-		if not bom_batches or not incomplete_level:
+		if not bom_batches or incomplete_level:
 			continue
 
```

Comment:
- We want to continue or skip processing there are **no batches to process** or **if all rows are completed**.
- Here `not incomplete_level` meant all levels are complete/there is no incomplete level.
- We don't want to skip if there is an incomplete level (which is happening now?), we want to process it

To simplify this, a newer snippet:
```py
all_complete = all(row.get("status") == "Completed" for row in bom_batches)
if not bom_batches or all_complete:
    continue
```